services/licenseStore.js

```diff
@@ -26,24 +26,56 @@ function buildPortedLicense(license, string, url) {
 }
 
 function buildLicensesIndex(licenses) {
-  return licenses.reduce((idx, license) => Object.assign(idx, { [license.id]: license }), {});
+  return licenses.reduce((idx, license) => {
+    Object.assign(idx, { [license.id]: license });
+    return idx;
+  }, {});
+}
+
+function buildLicenseNamesIndex(licenses) {
+  return licenses.reduce((idx, license) => {
+    const page = idx[license.name];
+    if (page) {
+      page.push(license.id);
+    } else {
+      Object.assign(idx, { [license.name]: [license.id] });
+    }
+    return idx;
+  }, {});
+}
```

Comment:
I'm not really happy with this ☝️ way of building an index, couldn't come up with a better solution, though. Improvement suggestions are very welcome 😄